### PR TITLE
LPS-201194, LPS-197895 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-194238 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-194238](https://liferay.atlassian.net/browse/LPS-194238) → [LPS-197895](https://liferay.atlassian.net/browse/LPS-197895)

# LPS-196842 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-196842](https://liferay.atlassian.net/browse/LPS-196842) → [LPS-201194](https://liferay.atlassian.net/browse/LPS-201194)
https://github.com/liferay-echo/liferay-portal/pull/14295

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/MenuItem.path [hunk 1]
```diff
@@ -50,6 +50,11 @@
 	<td>xpath=(//div | //li)[(contains(@class,'open') or contains(@class,'show')) and (contains(@class,'dropdown') or contains(.//*/@class,'dropdown'))]//*[self::a or self::span or self::button][normalize-space(text())='${key_menuItem}']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>DROPDOWN_MENU_ITEM_ACTIVE</td>
+	<td>//a[contains(@class, 'dropdown-item active')][contains(text(), '${key_menuItem}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/MenuItem.path:55]: `xpath=(//div | //li)[(contains(@class,'open') or contains(@class,'show')) and (contains(@class,'dropdown') or contains(.//*/@class,'dropdown'))]//*[self::a or self::span or self::button][contains(@class,'active')][normalize-space(text())='${key_menuItem}']`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 2]
```diff
@@ -846,4 +972,204 @@ definition {
...
+		}
+
+		task ("When filtering by name and sort is added") {
+			var baseURL = PropsUtil.get("portal.url");
+
+			Open.openInTheNewTab(value1 = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_users_admin_web_portlet_UsersAdminPortlet");
+
+			User.impersonateCP(
+				siteName = ${siteName},
+				userScreenName = "usersn");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:1031]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 3]
```diff
@@ -846,4 +972,204 @@ definition {
...
+		}
+
+		task ("When filtering by Display Page Template, ordering by Name and sort is added") {
+			var baseURL = PropsUtil.get("portal.url");
+
+			Open.openInTheNewTab(value1 = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_users_admin_web_portlet_UsersAdminPortlet");
+
+			User.impersonateCP(
+				siteName = ${siteName},
+				userScreenName = "usersn");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:1098]: Same as above

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Comment by rafaellajordao** [comment]
Hey @Tim-Cao , I've already made all the changes just waiting for Bea's response! :)

**Comment by Tim-Cao** [comment]
Btw, please rebase after https://github.com/liferay-echo/liferay-portal/pull/14282 merged. Thanks @rafaellajordao

**Comment by rafaellajordao** [comment]
> Btw, please rebase after #14282 merged. Thanks @rafaellajordao

Done!

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
Hi @rafaellajordao, I've splited cases related locked pages here. See changes https://github.com/liferay-echo/liferay-portal/pull/14333

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 4]
```diff
@@ -186,6 +186,77 @@ definition {
...
+		task ("When searching for a term on the web content admin page") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Web");
+
+			ManagementBar.setFilterAndOrder(orderBy = "Title");
+
+			AssertTextEquals.assertPartialText(
+				key_position = 1,
+				locator1 = "LexiconList#LIST_ENTRY_X",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:210]: Modify value of `LexiconList#LIST_ENTRY_X` to `xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 5]
```diff
@@ -186,6 +186,77 @@ definition {
...
+				key_position = 2,
+				locator1 = "LexiconList#LIST_ENTRY_X",
+				value1 = "A Web Content Title");
+		}
+
+		task ("The user can select the Reverse Sort Direction button via the keyboard") {
+			AssertElementAccessible(locator1 = "ManagementBar#ORDER_ICON_GENERIC");
+
+			KeyPress(
+				locator1 = "ManagementBar#ORDER_ICON_GENERIC",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:223]: Use the previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 6]
```diff
@@ -186,6 +186,77 @@ definition {
...
+				locator1 = "LexiconList#LIST_ENTRY_X",
+				value1 = "A Web Content Title");
+
+			AssertTextEquals.assertPartialText(
+				key_position = 2,
+				locator1 = "LexiconList#LIST_ENTRY_X",
+				value1 = "B Web Content Title");
+
+			KeyPress(
+				locator1 = "ManagementBar#ORDER_ICON_GENERIC",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:241]: Use the previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 7]
```diff
@@ -1957,6 +1957,133 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-196842. When the “Recent” filter is applied, the possibility of changing the order must be disabled"
+	@priority = 4
+	test ViewOrderButtonNotPresentWithRecentFilter {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1962]: ViewOrderButtonDisabledWhenFilterByRecent

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 8]
```diff
@@ -186,6 +186,77 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-196842. The user can select the Reverse Sort Direction button via the keyboard."
+	@priority = 4
+	test ViewAccessibilityInOrderButtonOfWebContentSearch {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:191]: RevertSortDirectionOfSearchResultsViaKeyboard

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 9]
```diff
@@ -1957,6 +1957,133 @@ definition {
...
+		}
+
+		task ("Then the possibility of changing the order must be disabled") {
+			AssertElementNotPresent(locator1 = "ManagementBar#ORDER_ICON_GENERIC");
+		}
+	}
+
+	@description = "This is a test for LPS-196842. When searched and filtered content exists, the order must be applied to the displayed content."
+	@priority = 4
+	test ViewOrderOfWebContentWithFilterAndSearch {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1987]: OrderByTitleInSearchResults

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 10]
```diff
@@ -1957,6 +1957,133 @@ definition {
...
+
+			AssertVisible(
+				key_menuItem = "Title",
+				locator1 = "MenuItem#DROPDOWN_MENU_ITEM_ACTIVE");
+		}
+	}
+
+	@description = "This is a test for LPS-196842. When varying filters or searching if order by is applied, the order by must be applied to the new search + filters."
+	@priority = 4
+	test ViewOrderOfWebContentWithFilterAndSearchVariation {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:2040]: OrderByDisplayDateInNewSearchResults

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/MenuItem.path [hunk 11]
```diff
@@ -50,6 +50,11 @@
 	<td>xpath=(//div | //li)[(contains(@class,'open') or contains(@class,'show')) and (contains(@class,'dropdown') or contains(.//*/@class,'dropdown'))]//*[self::a or self::span or self::button][normalize-space(text())='${key_menuItem}']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>DROPDOWN_MENU_ITEM_ACTIVE</td>
+	<td>xpath=(//div | //li)[(contains(@class,'open') or contains(@class,'show')) and (contains(@class,'dropdown') or contains(.//*/@class,'dropdown'))]//*[self::a or self::span or self::button][contains(@class,'active')][normalize-space(text())='${key_menuItem}']</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/MenuItem.path:57]: Keep this change for WebContent cases

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Would you mind modifying according to comments? Thanks! @rafaellajordao

**Pull request closed without merging**